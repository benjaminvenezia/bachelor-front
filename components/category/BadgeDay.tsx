import { View, Text, StyleSheet } from "react-native";

type Props = {
  dayText: string;
};

const BadgeDay = ({ dayText }: Props) => {
  return (
    <View style={styles.badge}>
      <Text style={styles.textBadge}>{dayText.substring(0, 1)}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  badge: {
    backgroundColor: "white",
    width: 15,
    height: 15,
    borderRadius: 10,
    textAlign: "center",
    alignItems: "center",
    justifyContent: "center",
    borderColor: "black",
  },
  textBadge: {
    fontWeight: "bold",
  },
});

export default BadgeDay;
