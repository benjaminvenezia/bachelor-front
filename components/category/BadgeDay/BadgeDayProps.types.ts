export type BadgeDayProps = {
  dayText: string;
};
