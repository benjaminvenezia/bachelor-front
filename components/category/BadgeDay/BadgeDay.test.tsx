import React from "react";
import renderer from "react-test-renderer";

import BadgeDay from "./BadgeDay";

describe("<BadgeDay />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<BadgeDay dayText="Lun" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
