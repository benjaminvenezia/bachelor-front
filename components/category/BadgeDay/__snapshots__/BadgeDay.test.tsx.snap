// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BadgeDay /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "black",
      "borderRadius": 10,
      "height": 15,
      "justifyContent": "center",
      "textAlign": "center",
      "width": 15,
    }
  }
>
  <Text
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    L
  </Text>
</View>
`;
